### ISSUE TITLE
Many UnhandledPromiseRejectionWarning on pause

### ISSUE THREAD

**Issue by mahnunchik** [OptimalBits/bull#934]
## Description

I've faced with a lot of messages like:

```
(node:38514) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 19)
(node:38514) UnhandledPromiseRejectionWarning: Error: Redis is already connecting/connected
    at Redis.<anonymous> (/tmp/node_modules/ioredis/lib/redis.js:252:14)
    at Promise._execute (/tmp/node_modules/bluebird/js/release/debuggability.js:303:9)
    at Promise._resolveFromExecutor (/tmp/node_modules/bluebird/js/release/promise.js:483:18)
    at new Promise (/tmp/node_modules/bluebird/js/release/promise.js:79:10)
    at Redis.connect (/tmp/node_modules/ioredis/lib/redis.js:250:10)
    at /tmp/node_modules/bull/lib/queue.js:1169:30
    at tryCatcher (/tmp/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/tmp/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/tmp/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/tmp/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/tmp/node_modules/bluebird/js/release/promise.js:693:18)
    at Async._drainQueue (/tmp/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/tmp/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/tmp/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:794:20)
    at tryOnImmediate (timers.js:752:5)
```

## Test code to reproduce

```js
await queue.pause(true);
```

## Bull version

bull@3.4.1

## Additional information

Node: v8.11.1

I'm using custom redis client:

```js
const Queue = require('bull');
const Redis = require('ioredis');

const client = new Redis(config.get('redis'));
const subscriber = new Redis(config.get('redis'));

const opts = {
  createClient: function(type) {
    switch(type) {
      case 'client':
        return client;
      case 'subscriber':
        return subscriber;
      default:
        return new Redis(config.get('redis'));
    }
  }
};

const queue = new Queue(name, opts);
```

**Comment by mahnunchik** [comment]
Yep, it solves the problem with a lot of messages. But why original error happened?

**Comment by electrikdevelopment** [comment]
For anyone else that runs into this. I found it in the following circumstances:

For me this was caused by queue.pause(true) being called multiple times on the same queue at the same time. 

In my case this was because I was catching many shutdown hooks and calling pause in the shutdown hook handler. In some circumstances I was getting two hooks invoked and therefore calling pause twice at the same time.
